### DISCUSSION TITLE
0.23 Docker Image / Modus / Bedeutung von Log-Eintrag unklar

### DISCUSSION THREAD

**Discussion by jaydee73** [wiggal/GEN24_Ladesteuerung#91] [Q&A]
Hi Wiggal,

ich habe gerade mein Docker-Image auf 0.23.1 hochgezogen. Erste kleine Hürde: Das Docker-Hub-Repo "gen24_ladesteuerung_alpine", welches ich ja bisher genutzt habe, wird anscheinend nicht mehr weiter gepflegt. Das war mir so bisher nicht bewusst. ;-) Daher schlug das Update des Containers auf den 0.23.1 erstmal fehl. Ok, geschwenkt auf eines der "neuen" Repos (php und apache) und dann ging es. Planst du, die beiden Repos tatsächlich parallel weiter zu pflegen?

Nun zur eigentlichen Frage:
In den ersten 5min-Aufrufen ist jetzt folgendes zu lesen:

`
***** BEGINN: 2024-08-10 10:45:01 *****

## MODBUS LADESTEUERUNG ###
aktuellePrognose: 3089
TagesPrognose-BattVollUm: 34411 - 18
PrognoseUberschuss/Stunde: 4205
Grundlast_Summe für Tag: 3625
aktuellePVProduktion/Watt: 5730
aktuelleEinspeisung/Watt: 5334
aktuelleBatteriePower/Watt: -12
GesamtverbrauchHaus/Watt: 384
aktuelleBattKapazität/Watt: 3921
Batteriestatus in Prozent: 46.8 %
LadewertGrund: Prognoseberechnung
Bisheriger Ladewert/Watt: 0
Bisheriger Ladewert/Prozent: 0.0 %
Neuer Ladewert/Watt: 432
Neuer Ladewert/Prozent: 5.62 %
newPercent_schreiben: 0

Änderung kleiner Schreibgrenze!

Fallback ist eingeschaltet.
InOutWRte_RvrtTms_Fallback: 18270
StorageControlMode: 3

In SQLite-Datei gespeichert!
***** ENDE: 2024-08-10 10:45:01 ***** 
`

Die Formulierung "Änderung kleiner Schreibgrenze" kannte ich so bisher noch nicht. Ist die neu? Bzw. was sagt sie genau aus? Da stehe ich etwas auf dem Schlauch.

Aufgrund der für heute guten Wetterprognose hätte ich jetzt folgenden Eintrag eher erwartet:

"Alte und Neue Werte unterscheiden sich weniger als die Schreibgrenzen des WR, NICHTS zu schreiben!! "

So wurde es auch vor dem Update auf 0.23 noch geloggt.

Ist das alles so erwartetes Verhalten? Du hattest ja geschrieben, dass du Modbus nicht getestet hast mit der 0.23, deswegen frage ich zur Sicherheit mal nach.

Dank dir,
JD

**Comment by jaydee73** [comment]
Ergänzung: Beim aktuellen Durchlauf sehe ich nun auch, dass er den StorageControlMode neu geschrieben hat:

`***** BEGINN: 2024-08-10 10:55:01 *****

## MODBUS LADESTEUERUNG ###
aktuellePrognose: 3278
TagesPrognose-BattVollUm: 33951 - 18
PrognoseUberschuss/Stunde: 4244
Grundlast_Summe für Tag: 3541
aktuellePVProduktion/Watt: 5405
aktuelleEinspeisung/Watt: 0
aktuelleBatteriePower/Watt: -5629
GesamtverbrauchHaus/Watt: -224
aktuelleBattKapazität/Watt: 3885
Batteriestatus in Prozent: 47.3 %
LadewertGrund: Prognoseberechnung
Bisheriger Ladewert/Watt: 0
Bisheriger Ladewert/Prozent: 0.0 %
Neuer Ladewert/Watt: 438
Neuer Ladewert/Prozent: 5.7 %
newPercent_schreiben: 0

Änderung kleiner Schreibgrenze!
StorageControlMode 3 neu geschrieben.

Fallback ist eingeschaltet.
InOutWRte_RvrtTms_Fallback: 18270
StorageControlMode: 3

In SQLite-Datei gespeichert!
***** ENDE: 2024-08-10 10:55:01 ***** `

Das ist für mich jetzt auch etwas verwunderlich, weil er doch vorher auch schon 3 was (siehe Eintrag 10.45 Uhr). Soll das so sein?

**Comment by jaydee73** [comment]
Update:
Da ich davon ausgehe, dass du kurz-/mittelfristig eh den Modbus-Teil in Rente schicken wirst, habe ich heute mal die Zeit genutzt, das Ganze auf http umzustellen. Aber weiterhin via Docker-Container (hier: Apache-Repo).

Läuft soweit, den Eintrag "Änderung kleiner Schreibgrenze!" habe ich allerdings weiterhin. Wobei es anscheinend "richtig" läuft, zumindest ist der Akku jetzt (18.45 Uhr) fast voll.

Eine Sache allerdings: Die grafische Darstellung (PV und QZ Bilanz) funktioniert nicht. logging steht auf 1, und in die sqlite wird auch geschrieben (sehe ich im Log und am Zeitstempel der Datei). Allerdings sind die beiden Tabs einfach leer. Weiße Seite, nix wird angezeigt. Alle anderen Tabs funktionieren, wenn ich die Seite mit IP:8880 aufrufe. Einfacherphpwebserver steht auf 0.

Hast du dazu eine Idee?

**Reply by wiggal** [reply]
> Update: Da ich davon ausgehe, dass du kurz-/mittelfristig eh den Modbus-Teil in Rente schicken wirst, habe ich heute mal die Zeit genutzt, das Ganze auf http umzustellen. Aber weiterhin via Docker-Container (hier: Apache-Repo).
> 
> Eine Sache allerdings: Die grafische Darstellung (PV und QZ Bilanz) funktioniert nicht. logging steht auf 1, und in die sqlite wird auch geschrieben (sehe ich im Log und am Zeitstempel der Datei). Allerdings sind die beiden Tabs einfach leer. Weiße Seite, nix wird angezeigt. Alle anderen Tabs funktionieren, wenn ich die Seite mit IP:8880 aufrufe. Einfacherphpwebserver steht auf 0.
> 
> Hast du dazu eine Idee?

Das mit der Grafik sehe ich mir später mal an

**Comment by jaydee73** [comment]
Moin @wiggal ,

so, nach der ersten Nacht folgende Erkenntnisse:

PV und QZ-Bilanz werden weiterhin nicht angezeigt.

Und zusätzlich fällt mir auf, dass das Script bereits jetzt am frühen Morgen wieder 200W in die Batterie schiebt, obwohl heute 50kw erwartet werden. Das ist mir gestern auch schon aufgefallen, aber ich dachte, ich warte mal den "Reset" in der Nacht ab. 

Anbei mal mein Log, falls du einen Blick werfen möchtest.

PS: Update auf die gerade neue 0.24 habe ich noch nicht gemacht.
[Crontab.log.zip](https://github.com/user-attachments/files/16571911/Crontab.log.zip)

**Reply by wiggal** [reply]
Hi, hab die Diskussion mal verschoben, da es im Wesentlichen ja die Steuerung betrifft.

**Reply by wiggal** [reply]
> Und zusätzlich fällt mir auf, dass das Script bereits jetzt am frühen Morgen wieder 200W in die Batterie schiebt, obwohl heute 50kw erwartet werden. Das ist mir gestern auch schon aufgefallen, aber ich dachte, ich warte mal den "Reset" in der Nacht ab.
> 
> Anbei mal mein Log, falls du einen Blick werfen möchtest.
> 
> PS: Update auf die gerade neue 0.24 habe ich noch nicht gemacht. [Crontab.log.zip](https://github.com/user-attachments/files/16571911/Crontab.log.zip)

Ja das mit den 200W ist in der 0.24 behoben und sieht dann so aus:

```
LadewertGrund: Ladewert 152 < Grenze_nachOben/2
Bisheriger Ladewert/Watt: 2200
Neuer Ladewert/Watt: 0

CHARGE_MAX geschrieben: 0W
```

**Reply by jaydee73** [reply]
Hi @wiggal,

nachdem das Script gestern (13.08.) vormittags nicht geschrieben hat und (wie geplant) erst am Nachmittag die Batterie wieder langsam vollgeladen hat, hat es heute sofort wieder ab dem frühen Morgen ca. 200W reingeladen, bis jetzt (15.30 Uhr) unverändert. Ist die Version 0.24. Log könnte ich später hochladen, wenn dir das was hilft.

**Reply by wiggal** [reply]
ja bitte die Log posten

**Reply by jaydee73** [reply]
here we go.
[Crontab.log.zip](https://github.com/user-attachments/files/16615820/Crontab.log.zip)

**Reply by wiggal** [reply]
Gestern:
```
***** BEGINN:  2024-08-13 05:51:01 *****
Batteriestatus in Prozent:   63.9 %
LadewertGrund:  Ladewert 175 < Grenze_nachOben/2
Bisheriger Ladewert/Watt:    3000
Neuer Ladewert/Watt:         0
CHARGE_MAX geschrieben: 0W
```
Heute:
```
***** BEGINN:  2024-08-14 05:51:01 *****
Batteriestatus in Prozent:   45.9 %
LadewertGrund:  Prognoseberechnung
Bisheriger Ladewert/Watt:    3000
Neuer Ladewert/Watt:         262
CHARGE_MAX geschrieben: 262W
```
Wie im LadewertGrund gestern stand, ist der Ladewert beim ersten Schreiben (Produktion erstmals über 10 Watt!) 175W.
Der Wert ist so niedrig, da der Akku noch 63.9 % hat, liegt unter Grenze_nachOben/2 (vermtl. 450/2) und wird dadurch auf 0 gesetzt.
Heute war der Akku zu dem Zeitpunkt bei 45.9 %, deshalb liegt der erste Ladewert bei 262W, was über Grenze_nachOben/2 ist und damit nicht auf 0 gesetzt wird.

**Reply by wiggal** [reply]
was hast du den für einer BatSparFaktor?

**Reply by jaydee73** [reply]
Ok, Danke für die Recherche. Auch wenn es also vermutlich erwartetes Verhalten ist, so richtig logisch finde ich es nicht. Der Akku hatte noch 46%, was ca. 3,5kwh entspricht. Sprich ca. 4kwh könnte die Batterie noch aufnehmen. Die Prognose für den Tag sagte über 40kw voraus. Klar kann da den ganzen Tag lang 200W reingeschoben werden und abends nochmal ein bisschen mehr. Ich hatte so ein Verhalten nur bisher nie gehabt (oder zumindest nicht gesehen). Da passierte an solchen Tagen bis 14 Uhr nix und dann ging es langsam los. Klar, im Ergebnis läuft beides auf das Gleiche raus.

Ja, Grenze nach oben ist bei mir 450, richtig vermutet. Dann werde ich mich mit meinem inneren Monk mal damit arrangieren, dass das so sein soll. ;-) Ich wüsste auch aktuell nicht, mit welchen Parameter-Änderungen ich darauf Einfluss nehmen könnte. Vergrößerung der Schreibgrenze würde das ja auch nur etwas herauszögern, oder nicht?

**Reply by jaydee73** [reply]
BatSpar ist bei 0.8

**Reply by wiggal** [reply]
es gibt einige Faktoren die das Ladeverhalten beeinflussen:
Einspeisebegrenzung, BatSparFaktor, Schreibgenzenachoben, Akkustand, ACLeistung und Prognose.

Die Frage ist, was du erreichen willst.
Eine gleichmäßige Ladung oder Reservierung für die Mittagszeit z.B. wegen der Einspeisebegrenzung...

**Reply by jaydee73** [reply]
Na ja, ich war nur irritiert, weil für mich sind 64% und 46% SoC gefühlt gleich, wenn der Tag noch genug Sonne bringt. Trotzdem agiert das Script anders. Vielleicht/vermutlich ist ja auch eine gleichmäßige, langsame und niedrige Ladung besser für die Akkus als bis 14 Uhr nix zu machen und dann mehr reinzupumpen.

Einspeisebegrenzung ist bei mir normalerweise kein Argument. Ich habe ne Ost/West-Ausrichtung und eh seltenst einen vollen Peak. Und nen E-Auto habe ich auch nicht, was berücksichtigt werden sollte.

Egal. Es ist so wie es ist, das Script arbeitet wie erwartet und ich gewöhne mich dran. Alles gut.

Dank dir für die Unterstützung.

**Comment by wiggal** [comment]
> Planst du, die beiden Repos tatsächlich parallel weiter zu pflegen?

Ja werd vorerst beide erstellen.

> 
> "Alte und Neue Werte unterscheiden sich weniger als die Schreibgrenzen des WR, NICHTS zu schreiben!! "
> 

Wurde durch `Änderung kleiner Schreibgrenze!` ersetzt.

**Reply by wiggal** [reply]
Kannst du bitte mal ein Bild von einem Grafiktab schicken

**Reply by jaydee73** [reply]
Kein Problem. Gibt aber nicht viel zu sehen. ;-)
<img width="1315" alt="Bildschirmfoto 2024-08-11 um 10 08 45" src="https://github.com/user-attachments/assets/de2a02ec-3243-4c73-8db0-e0b03695a87a">

**Reply by wiggal** [reply]
Habs bei mir mal mir der apache_docker-compose.yml probiert, läuft.
Hast du die CONFIG/default_priv.ini angelegt und gemapped?
Hast du die aktuelle apache_docker-compose.yml?

**Reply by jaydee73** [reply]
Ich würde beide Fragen mit ja beantworten. Habe jetzt allerdings nochmal einen komplett neuen Container mit der 0.24 gemacht (Apache). Nun sehe ich auch Grafiken in den Tabs. Keine Ahnung, was da vorher quer hing. Nun geht es. Sorry für die Verwirrung.

Dann schaue ich mir jetzt nochmal den Tag über die Ladewerte an, ob er erst mal bei 0 bleibt...

Danke schon mal für deine Hilfe.

**Reply by jaydee73** [reply]
Kleiner Hinweis noch:
Ab der 0.24 muss dann ja bei der Docker-Variante auch die Prog_Steuerung.sqlite gemapped werden, oder ist die statisch?
Wenn Mapping notwendig/sinnvoll ist, müsstest du das in die Readme bei Github noch mit aufnehmen.

Wobei, wenn ich so drüber nachdenke...vermutlich ist die Datei statisch und muss gar nicht gemapped werden. Na ja, kannst du ja gerne auch nochmal bestätigen. Dachte, weil sie im CONFIG-Ordner liegt, dass sie wie die anderen Dateien dort auch eben nicht statisch wäre.

**Reply by wiggal** [reply]
Ja, ist nicht viel kaputt, wenn sie nicht gemapped ist, wäre aber besser wenn doch.
Ich baue sie noch ein, danke für den Hinweis.
